fqdn/__init__.py

```diff
@@ -26,15 +26,23 @@ class FQDN:
     length of a label is 63 bytes without the leading length byte.
     """
 
-    FQDN_REGEX = re.compile(
+    STRICT_FQDN_REGEX = re.compile(
         r"^((?!-)[-A-Z\d]{1,63}(?<!-)\.)+(?!-)(?=.*[A-Z])([-A-Z\d]{1,63})?(?<!-)\.?$",
         re.IGNORECASE,
     )
+    LOOSE_FQDN_REGEX = re.compile(
+        r"^((?![-_])[-_A-Z\d]{1,63}(?<!-)\.)*((?!-)[-A-Z\d]{1,63}(?<!-)\.)(?!-)(?=.*[A-Z])([-A-Z\d]{1,63})?(?<!-)\.?$",
+        re.IGNORECASE,
+    )
```

Comment:
cpython caches regexes. It would be nice to just store the string constants here, and call re.compile on line 43.